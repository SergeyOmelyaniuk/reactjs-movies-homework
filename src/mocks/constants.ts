import { Category, Genre, Language } from '../types';

export const categories: Category[] = [
	{ id: 1, value: 'popular', title: 'Popular' },
	{ id: 2, value: 'top_rated', title: 'Top rated' },
	{ id: 3, value: 'upcoming', title: 'Upcoming' },
];

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const genres: Genre[] = [
	{ id: 1, name: 'Genre 1' },
	{ id: 2, name: 'Genre 2' },
	{ id: 3, name: 'Genre 3' },
	{ id: 4, name: 'Genre 4' },
	{ id: 5, name: 'Genre 5' },
];
