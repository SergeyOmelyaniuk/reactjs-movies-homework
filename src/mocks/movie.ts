import { MovieDetails } from './../types';

export const movieDescription: MovieDetails = {
	poster_path: 'film_1.jpg',
	genres: [{ id: 1, name: 'first genre' }],
	title: 'Красотка на взводе',
	release_date: '2021-05-26',
	vote_average: 68,
	revenue: 120000,
	runtime: 120,
	overview:
		'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
};
