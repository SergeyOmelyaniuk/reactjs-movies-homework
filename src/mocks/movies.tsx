import { Movie } from '../types';

export const movies: Movie[] = [
	{
		id: 1,
		title: 'Красотка на взводе',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_1.jpg',
		vote_average: 68,
		popularity: 0.5,
		revenue: 120000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [
			{
				name: 'Kate Beckinsale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_1.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Lindy',
			},
			{
				name: 'Stanley Tucci',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_2.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Dr. Ivan Munchin',
			},
			{
				name: 'Bobby Cannavale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_3.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Vicars',
			},
			{
				name: 'Jai Courtney',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_4.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Justin',
			},
			{
				name: 'Laverne Cox',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_5.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Nevin',
			},
			{
				name: 'David Bradley',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_6.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Gareth Fizel',
			},
		],
	},
	{
		id: 2,
		title: 'Сокровище Счастливчика',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_2.jpg',
		vote_average: 72,
		popularity: 0.5,
		revenue: 230000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 3,
		title: 'Сталинград',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_3.jpg',
		vote_average: 76,
		popularity: 0.5,
		revenue: 240000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 4,
		title: 'Люди в черном: Интернэшнл',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_4.jpg',
		vote_average: 69,
		popularity: 0.5,
		revenue: 136000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 5,
		title: 'Значит, война',
		release_date: '2021-05-26',
		genre: ['Боевик'],
		img: 'film_5.jpg',
		vote_average: 96,
		popularity: 0.5,
		revenue: 226000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
];
