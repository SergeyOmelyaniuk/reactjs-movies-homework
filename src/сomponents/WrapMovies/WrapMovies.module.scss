.wrapMovies {
	display: flex;
	justify-content: space-between;
	flex-wrap: wrap;
	margin: 0 -40px;
}

.wrapMovie {
	margin: 0px 40px 40px 40px;
}
