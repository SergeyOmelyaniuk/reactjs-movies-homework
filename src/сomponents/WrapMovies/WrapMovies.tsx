import React from 'react';
// import { Movie } from '../../types';
import MovieCard from '../MovieCard';
import styles from './WrapMovies.module.scss';

interface WrapMoviesProps {
	//TODO
	// movies: Movie[];
	movies: any;
}

function WrapMovies(props: WrapMoviesProps) {
	return (
		<div className={styles.wrapMovies}>
			{props.movies.map((movie: any) => {
				return (
					<div key={movie.id} className={styles.wrapMovie}>
						<MovieCard movie={movie} />
					</div>
				);
			})}
		</div>
	);
}

export default WrapMovies;
