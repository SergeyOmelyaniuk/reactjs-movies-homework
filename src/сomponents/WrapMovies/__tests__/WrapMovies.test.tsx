import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import WrapMovies from '..';
import { movies } from '../../../mocks/movies';

describe('WrapMovies', () => {
	test('should render', () => {
		const { container } = render(
			<BrowserRouter>
				<WrapMovies movies={movies.popular} />
			</BrowserRouter>
		);

		expect(container).toMatchSnapshot();
	});
});
