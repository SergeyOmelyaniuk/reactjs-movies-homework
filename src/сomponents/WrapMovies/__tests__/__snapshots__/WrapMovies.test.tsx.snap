// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WrapMovies should render 1`] = `
<div>
  <div
    class="wrapMovies"
  >
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            68
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Красотка на взводе
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
            <span>
              Комедия
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            72
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Сокровище Счастливчика
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
            <span>
              Комедия
            </span>
            <span>
              Драма
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            76
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Сталинград
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
            <span>
              Комедия
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            69
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Люди в черном: Интернэшнл
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
            <span>
              Комедия
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            96
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Значит, война
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            77
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Приключения желтого чемоданчика
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            79
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Поезд в Пусан
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            79
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Никто не выжил
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            53
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Ведьма
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            53
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Звёздные войны: Пробуждение силы
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            58
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Марсианин
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            100
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Республика Z
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            90
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Планета 51
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            86
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Красные туфельки и семь гномов
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            92
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Капитошка
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            58
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Война будущего
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            86
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Чернобыль
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            81
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Странный дом
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            79
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Дальний космос
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="wrapMovie"
    >
      <div
        class="movieCard"
      >
        <div
          class="wrapImage"
        >
          <div
            class="overlay"
          />
          <img
            alt=""
            class="image"
          />
          <span
            class="rating"
          >
            20
          </span>
          <div
            class="preview"
          />
        </div>
        <div
          class="content"
        >
          <h2
            class="title"
          >
            Мандалорец
          </h2>
          <p
            class="genre"
          >
            <span>
              Боевик
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
